apps/mobile-wallet/src/screens/Addresses/AddressesScreen.tsx

```diff
@@ -1,107 +1,48 @@
-  const defaultAddress = useAppSelector(selectDefaultAddress)
-  const [selectedAddressHash, setSelectedAddressHash] = useState(defaultAddress?.hash ?? '')
-  const selectedAddress = useAppSelector((s) => selectAddressByHash(s, selectedAddressHash))
 
-  const [heightCarouselItem, setHeightCarouselItem] = useState(235)
-  const [scrollToCarouselPage, setScrollToCarouselPage] = useState<number>()
-  const [isSwiping, setIsSwiping] = useState(false)
-
-  useEffect(() => {
-    if (defaultAddress?.hash) {
-      setSelectedAddressHash(defaultAddress.hash)
-      setScrollToCarouselPage(0)
-    }
-  }, [addressHashes, defaultAddress?.hash])
-
-  const onAddressCardsScrollEnd = (index: number) => {
-    Haptics.selectionAsync()
-
-    if (index < addressHashes.length) {
-      setSelectedAddressHash(addressHashes[index])
-      setScrollToCarouselPage(index)
-      setIsSwiping(false)
-    }
-  }
-
-  const handleAddressSelect = (addressHash: AddressHash) => {
-    setSelectedAddressHash(addressHash)
-    setScrollToCarouselPage(addressHashes.findIndex((hash) => hash === addressHash))
-    sendAnalytics({ event: 'Used address quick navigation' })
+  const handleAddressPress = (addressHash: AddressHash) => {
+    dispatch(openModal({ name: 'AddressDetailsModal', props: { addressHash } }))
   }
 
-  const openAddressSelectModal = () =>
-    dispatch(openModal({ name: 'SelectAddressModal', props: { onAddressPress: handleAddressSelect } }))
-
-  const renderAddressCard = ({ item }: { item: string }) => (
-    <View onLayout={(event) => setHeightCarouselItem(event.nativeEvent.layout.height + 15)} key={item}>
-      <AddressCard
-        addressHash={item}
-        onSettingsPress={() => navigation.navigate('EditAddressScreen', { addressHash: item })}
-      />
-    </View>
-  )
-
-  if (!selectedAddress) return null
-
   return (
     <BottomBarScrollScreen
       refreshControl={<RefreshSpinner progressViewOffset={190} />}
       hasBottomBar
       contentPaddingTop
       {...props}
     >
-      <Content style={contentStyle}>
-        <Carousel
-          data={addressHashes}
-          renderItem={renderAddressCard}
-          onSwipingStart={() => setIsSwiping(true)}
-          onScrollEnd={onAddressCardsScrollEnd}
-          padding={20}
-          distance={10}
-          height={heightCarouselItem}
-          scrollTo={scrollToCarouselPage}
-          FooterComponent={
-            addresses.length > 2 && (
-              <Button onPress={openAddressSelectModal} iconProps={{ name: 'list' }} squared compact />
-            )
-          }
-        />
-        {selectedAddress && <AddressesTokensList addressHash={selectedAddress.hash} isRefreshing={isSwiping} />}
+      <Content>
+        {addresses.map((address, i) => (
+          <AddressBox
+            key={address.hash}
+            addressHash={address.hash}
+            isLast={i === addresses.length - 1}
+            onPress={() => handleAddressPress(address.hash)}
+          />
+        ))}
```

Comment:
We should use a FlashList here, like we do in all other places where we render lists of addresses

Reply:
I decided not to for UI consistency and simplicity sake. FlashList don't support overflow, so the gradient fading effect we have in other screens would be hard to reproduce. Also, we know for a fact that very few users have more that 10 addresses, so the perfs benefit is super small. 
I suggest we give it a shot in an upcoming PR.

Reply:
Hm, ok. I hear you about UI consistency but then we don't have code consistency.

I have concerns about this, because each address row is an "expensive" component with lots of Tanstack queries and subcomponents. Is it really worth it removing this optimization for the sake of `overflow`? Dunno. BTW, which gradient are you referring to? I don't see any gradient in the list of addresses in your branch.

I'm ok to proceed like this. However, I would def add a TODO in the code explaining your logic (I won't remember this discussion in the near future, I am sure!) and why we switch from flashlist to fully rendered list. I would also add a subtask in this issue:
- https://github.com/alephium/alephium-frontend/issues/19

Reply:
Will be improved soon as discussed: https://github.com/alephium/alephium-frontend/issues/1111